PIL/EpsImagePlugin.py

```diff
@@ -330,10 +330,13 @@ def _find_offset(self, fp):
 
         return (length, offset)
 
-    def load(self, scale=1):
+    def load(self, scale=1, size=None):
         # Load EPS via Ghostscript
         if not self.tile:
             return
+        # Override the size of the image if specified
+        if not size is None:
+            self.size = size
         self.im = Ghostscript(self.tile, self.size, self.fp, scale)
         self.mode = self.im.mode
         self.size = self.im.size
```

Comment:
Any change that your `self.size = size` line makes would be reverted by this line, yes?

In which case, perhaps there is also an issue with your test, since it is passing anyway.